src/lib.rs

```diff
@@ -81,6 +81,15 @@ pub struct Id {
     pub index: u32,
 }
 
+/// This represents a globally unique crate identifier, which should allow
+/// for differentiation between different crate targets or versions and should
+/// point to the same crate when pulled by different other, dependent crates.
+#[derive(Debug, Clone, RustcDecodable, RustcEncodable, PartialEq, Eq, Hash)]
+pub struct GlobalCrateId {
+    pub name: String,
+    pub disambiguator: String,
```

Comment:
Could you document what the disambiguator looks like please? What are the valid values?

Reply:
@nrc so the disambiguator was changed internally from a hex-string to a 128 bit number (`Fingerprint(u64, u64)`, [here](https://github.com/rust-lang/rust/pull/45476)), as it was effectively a hex-formatted 128 bit value in a string.
Should I change it to use a `u128` or `(u64, u64)` here?

Reply:
I couldn't see where Fingerprint was defined. Ideally, you would use FingerPrint here, but that would probably mean pulling it into rls-data which might not be practical. If FingerPrint is defined as (u64, u64) then I'd use that, otherwise u128.